### DISCUSSION TITLE
DuckDB CLI: environmental variable as parameter

### DISCUSSION THREAD

**Discussion by frafra** [duckdb/duckdb#8350] [Q&A]
Hi!
It would be nice to have a query that can be parametrized using environmental variables or command line parameters. 
That would allow executing queries such as `select * from table where column = $var`. It does not seem to be possible, without using `duckdb << EOF`, but such approach does not escape values properly.

Any idea?

Thanks in advance :)

**Comment by orzom411** [comment]
An example might work for you:

`.\duckdb :memory: "SELECT '%PATH%' as the_answer"`

https://duckdb.org/docs/api/cli.html

The example there was:
`./duckdb :memory: "SELECT 42 as the_answer"`

Hope that helps.

**Reply by frafra** [reply]
Thank you! This is the Windows equivalent of the shell variable substitution I was mentioning, and it works well only if the input is considered to be safe, and it is manually escaped (for example, it should not contain `;` or `--`): otherwise the effect would be similar to a SQL-injection.

**Comment by orzom411** [comment]
I picked "PATH" as it has the semi, it wasn't an issue, a little more effort would be needed.  Limit, escape single quote, I'm thinking is an issue with this method but I wasn't able to recall if I could set an environment value with a single quote.  

Anyway, here was the results of my quick test.

```
C:\temp\2023\07\20230724>\tools\duckdb :memory: "SELECT '%PATH%' as the_answer"
┌──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┐
│                                                      the_answer                                                      │
│                                                       varchar                                                        │
├──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┤
│ C:\Program Files (x86)\Common Files\Oracle\Java\javapath;C:\Program Files\Microsoft MPI\Bin\;C:\WINDOWS\system32;C.  │
└──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┘
```

I agree it would be useful to have access to the full environment and I thought it was possible, thinking of the S3 credentials, sorry my suggestion didn't cover it for you.

**Reply by orzom411** [reply]
PS, confirmed single quote issue:

```
set JUSTTHINKING="Odd's Are Good"

\tools\duckdb :memory: "SELECT '%JUSTTHINKING%' as the_answer"
┌─────────┐
│    s    │
│ varchar │
├─────────┤
│ Odd     │
└─────────┘
Error: Parser Error: syntax error at or near "Are"
LINE 1: Are
```

**Comment by mskyttner** [comment]
@frafra would this be similar to https://github.com/duckdb/duckdb/issues/4038?

**Reply by frafra** [reply]
Yes, that is it!
